apps/src/aiTutor/views/AITutorChatWorkspace.tsx

```diff
@@ -13,44 +13,79 @@ import WarningModal from './WarningModal';
 import style from './ai-tutor.module.scss';
 
 const AITutorChatWorkspace: React.FunctionComponent = () => {
+  const conversationContainerRef = useRef<HTMLDivElement>(null);
   const storedMessages = useAppSelector(state => state.aiTutor.chatMessages);
   const isWaitingForChatResponse = useAppSelector(
     state => state.aiTutor.isWaitingForChatResponse
   );
+  const showSuggestedPrompts = useAppSelector(
+    state => state.aiTutor.showSuggestedPrompts
+  );
 
-  const showWaitingAnimation = () => {
-    if (isWaitingForChatResponse) {
-      return (
-        <img
-          src="/blockly/media/aichat/typing-animation.gif"
-          alt={'Waiting for response'}
-          className={style.waitingForResponse}
-        />
-      );
-    }
-  };
+  const [feedbackDetailsOpen, setFeedbackDetailsOpen] = useState(false);
+  useEffect(() => {
+    // Autoscroll to the bottom of the workspace when new messages, suggested prompts,
+    // or waiting animation is displayed.
+    setTimeout(() => {
+      if (conversationContainerRef.current) {
+        conversationContainerRef.current.scrollTo({
+          top: conversationContainerRef.current.scrollHeight,
+          behavior: 'smooth',
+        });
+      }
+    }, 250); // Small delay to ensure DOM updates before scrolling.
```

Comment:
Increased from `100` to `250` to allow for suggested prompts to update.

Reply:
Why does it take that long for the suggested prompts to update?

Reply:
I'm not sure. I updated this when I saw on the adhoc that:
- I submitted a couple messages already,
- the program was in a state where there were no errors,
- I added a program error and  ran the code and suggested prompts were displayed.
At this point, I saw that the scrolling was not being updated (the suggested prompts were half-hidden). The other messages and the assistant message details was working as expected.
Thus, I updated the delay and went through steps above, and scrolling happened as expected.

Reply:
Is there a way that we could trigger this effect once the suggested prompts appear rather than doing an arbitrary wait (can be a follow-up)?

Reply:
Maybe related to steps in `getSuggestedPromptOptionsByLabType` in `AITutorSuggestedPrompts.tsx`. I wonder if `hasPythonlabError` and `validationState.satisfied` take more time to send final value after user clicks on 'Run' or 'Validate'?

Reply:
Yeah, because we have to wait for the program to run and to parse the output.

Reply:
> Is there a way that we could trigger this effect once the suggested prompts appear rather than doing an arbitrary wait (can be a follow-up)?

I will investigate in the fast follow-up. Thanks!